connectors/tests/test_job_cleanup.py

```diff
@@ -0,0 +1,84 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+
+import asyncio
+from unittest.mock import AsyncMock, Mock, call, patch
+
+import pytest
+
+from connectors.services.job_cleanup import STUCK_JOB_ERROR, JobCleanUpService
+from connectors.tests.test_byoc import AsyncIterator
+
+CONFIG = {
+    "elasticsearch": {
+        "host": "http://nowhere.com:9200",
+        "user": "elastic",
+        "password": "changeme",
+    },
+    "service": {
+        "max_errors": 20,
+        "max_errors_span": 600,
+        "job_cleanup_interval": 1,
+    },
+    "native_service_types": ["mongodb"],
+}
+
+
+def create_service():
+    return JobCleanUpService(CONFIG)
+
+
+def mock_connector():
+    connector = Mock()
+    connector.id.return_value = "1"
+    connector._sync_done = AsyncMock()
+    return connector
+
+
+def mock_sync_job():
+    job = Mock()
+    job.job_id.return_value = "1"
+    job.index_name.return_value = "index_name"
+    job.connector_id.return_value = "1"
+    return job
+
+
+@pytest.mark.asyncio
+@patch("connectors.byoc.SyncJobIndex.delete_jobs")
+@patch("connectors.byoc.SyncJobIndex.delete_indices")
+@patch("connectors.byoc.SyncJobIndex.stuck_jobs")
+@patch("connectors.byoc.SyncJobIndex.orphaned_jobs")
+@patch("connectors.byoc.ConnectorIndex.fetch_by_id")
+@patch("connectors.byoc.ConnectorIndex.supported_connectors")
+@patch("connectors.byoc.ConnectorIndex.all_connectors")
```

Comment:
Not sure if there's any better way to patch so many methods.

Reply:
It depends on what you do with `SyncJobIndex` `ConnectorIndex` 

If you don't care about thee, you could try to pass a magic mock for those

Reply:
Do you mean mock `ConnectorIndex` and `SyncJobIndex` directly?

like 

```python
@patch("connectors.byoc.SyncJobIndex")
@patch("connectors.byoc.ConnectorIndex")
async def test_cleanup_jobs(
    connector_index,
    sync_job_index
):
    connector_index.all_connectors.return_value = AsyncGeneratorFake([connector])
    ...
````